Added my changes to the JavaScript.md file

maxthewang opened the pull request:
My contributions added to the pre-existing JavaScript.md file. I wrote sections which describe:
- how to embed JS in HTML
- how to link to a .js file in an HTML file
- examples which demonstrate basic JS functionality
as well as one more helpful resource in the Additional Resources section (the CS50 course)

KyojiGoto reviewed (commented):
https://developer.mozilla.org/en-US/docs/Web/JavaScript

MDN also has great resources for everything JavaScript related, it may be useful to add that to the resources list

KyojiGoto commented on Topics/Tech_Stacks/JavaScript.md:
```diff
@@ -1,11 +1,140 @@
+
+## Introduction: Embedding JavaScript in an HTML File
+
+This section assumes you are already familiar with the basics of HTML. For an introduction of how one can integrate JavaScript into web development, we will explore the basic scenario of embedding JavaScript code in HTML.
```
Adding a link to resources to learn html might be useful here as well

KyojiGoto reviewed (commented):
Providing screenshots of how the given html files will render and the overall file structure might be useful as well.

maxthewang commented:
> https://developer.mozilla.org/en-US/docs/Web/JavaScript
> 
> MDN also has great resources for everything JavaScript related, it may be useful to add that to the resources list

Great resource, I briefly came across it while writing the examples and should've included it.

> Adding a link to resources to learn html might be useful here as well

That's a good idea. I'll check if the most up-to-date version of the repo already has an article written about it, or I'll seek out a good one on the internet otherwise.

> Providing screenshots of how the given html files will render and the overall file structure might be useful as well.

Thanks, I forgot I could add images in Markdown, that'll definitely help the reader understand the examples better.

maxthewang commented:
Added changes requested in feedback (images, additional resources), also added table of contents and link to HTML guide on Tech Stacks page so I could include it in resources.

KyojiGoto commented:
Everything looks great, the pictures really tie the article together. This PR looks ready to merge.
